goldigen/go_path_checker_test.go

```diff
@@ -36,42 +46,59 @@ var _ = Describe("GoPathChecker", func() {
 		})
 
 		It("should return an empty string if the given output path is empty", func() {
-			checker := main.NewGoPathChecker(verbose)
+			checker := NewGoPathChecker(verbose)
 			Expect(checker.PackageName("")).To(BeEmpty())
 		})
 
 		It("should return an empty string if the given output path is not inside the $GOPATH", func() {
-			checker := main.NewGoPathChecker(verbose)
+			checker := NewGoPathChecker(verbose)
 			Expect(checker.PackageName("/usr/lib/go/src/foo/bar/baz/goldi.go")).To(BeEmpty())
 		})
 
 		It("should return the correct package name if the given output path is inside $GOPATH", func() {
-			checker := main.NewGoPathChecker(verbose)
+			checker := NewGoPathChecker(verbose)
 			Expect(checker.PackageName("/home/fgrosse/go/src/foo/bar/baz/goldi.go")).To(Equal("foo/bar/baz"))
 		})
 
+		It("should panic if cannot get absolute file path", func() {
+			filepathAbs = func(path string) (string, error) {
+				return "", errors.New("err")
+			}
+			checker := NewGoPathChecker(verbose)
+			Expect(func() { checker.PackageName("/home/fgrosse/go/src/foo/bar/baz/goldi.go") }).To(Panic())
+			filepathAbs = filepath.Abs
+		})
```

Comment:
I don't think this test case justifies introducing the global variable `filepathAbs`. I would rather avoid this extra indirection and complexity than have this code branch tested.

Reply:
I can revert this with no problems, but that's a private global variable. It's the same situation as for ioutil.ReadAll case mentioned above.